app/controllers/application_controller.rb

```diff
@@ -5,6 +5,147 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+       redirect "/tweets"
+    else
+      erb :"/users/create_user"
+    end
+  end
+
+  post '/signup' do
+    if !params[:username].empty? && !params[:email].empty? &&!params[:password].empty?
+      @user = User.create(params)
+      session[:id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"/users/login"
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+
+    if user && user.authenticate(params[:password])
+      session[:id] = user.id
+      redirect "/tweets"
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = User.find(session[:id])
+      erb :"/tweets/tweets"
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect '/login'
+    end
+
+    redirect '/login'
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find(session[:id])
+      erb :"/tweets/create_tweet"
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/users/:user_id' do
+    if logged_in?
+      @user = User.find(params[:user_id])
+      if !params[:content].empty?
+        @user.tweets << Tweet.create(content: params[:content])
```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```